packages/devtools_server/lib/src/handlers.dart

```diff
@@ -103,3 +177,182 @@ class ServerApi {
+        Platform.operatingSystem == 'windows' ? 'APPDATA' : 'HOME';
+    final String value = Platform.environment[envKey];
+    return value == null ? '.' : value;
+  }
+
+  IOPersistentProperties properties;
+
+  void reset() {
+    properties.remove('firstRun');
+    properties['enabled'] = false;
+    properties['surveyShownCount'] = 0;
+    properties['surveyActionTaken'] = false;
+  }
+
+  bool get isFirstRun {
+    properties['firstRun'] = properties['firstRun'] == null;
+    return properties['firstRun'];
+  }
+
+  bool get enabled {
+    if (properties['enabled'] == null) {
+      properties['enabled'] = false;
+    }
+
+    return properties['enabled'];
+  }
+
+  set enabled(bool value) {
+    properties['enabled'] = value;
+    return properties['enabled'];
+  }
+
+  int get surveyShownCount {
+    if (properties['surveyShownCount'] == null) {
+      properties['surveyShownCount'] = 0;
+    }
+
+    return properties['surveyShownCount'];
+  }
+
+  void incrementSurveyShownCount() {
+    surveyShownCount; // Insure surveyShownCount has been initialized.
+    properties['surveyShownCount'] += 1;
+  }
+
+  bool get surveyActionTaken => properties['surveyActionTaken'] == true;
+
+  set surveyActionTaken(bool value) {
+    properties['surveyActionTaken'] = value;
+  }
+}
+
+abstract class PersistentProperties {
+  PersistentProperties(this.name);
+
+  final String name;
+
+  dynamic operator [](String key);
+  void operator []=(String key, dynamic value);
+
+  /// Re-read settings from the backing store. This may be a no-op on some
+  /// platforms.
+  void syncSettings();
+}
+
+const JsonEncoder _jsonEncoder = JsonEncoder.withIndent('  ');
+
+class IOPersistentProperties extends PersistentProperties {
+  IOPersistentProperties(String name, {String documentDirPath}) : super(name) {
+    String fileName = '.${name.replaceAll(' ', '_')}';
+    documentDirPath ??= DevToolsUsage.userHomeDir();
+    _file = File(path.join(documentDirPath, fileName));
+    if (!_file.existsSync()) {
+      _file.createSync();
+    }
+    syncSettings();
+  }
+
+  IOPersistentProperties.fromFile(File file) : super(path.basename(file.path)) {
+    _file = file;
+    if (!_file.existsSync()) {
+      _file.createSync();
+    }
+    syncSettings();
+  }
+
+  File _file;
+  Map _map;
+
+  @override
+  dynamic operator [](String key) => _map[key];
+
+  @override
+  void operator []=(String key, dynamic value) {
+    if (value == null && !_map.containsKey(key)) return;
+    if (_map[key] == value) return;
+
+    if (value == null) {
+      _map.remove(key);
+    } else {
+      _map[key] = value;
+    }
+
+    try {
+      _file.writeAsStringSync(_jsonEncoder.convert(_map) + '\n');
+    } catch (_) {}
+  }
+
+  @override
+  void syncSettings() {
+    try {
+      String contents = _file.readAsStringSync();
+      if (contents.isEmpty) contents = '{}';
+      _map = jsonDecode(contents);
+    } catch (_) {
+      _map = {};
+    }
+  }
+
+  remove(String propertyName) {
```

Comment:
add void return type to method signature?

Reply:
Done.